library/lib.ml

```diff
@@ -328,16 +335,17 @@ let start_compilation s mp =
   lib_state := { !lib_state with comp_name = Some s;
                                  path_prefix = prefix }
 
+let open_blocks_message es =
+  let open_block_name = function
+      | oname, OpenedSection _ -> str "section " ++ pr_id (basename (fst oname))
+      | oname, OpenedModule (ty,_,_,_) -> str (module_kind ty) ++ spc () ++ pr_id (basename (fst oname))
+      | _ -> assert false in
+  str "The " ++ pr_enum open_block_name es ++ str " need to be closed."
```

Comment:
There is `CString.conjugate_verb_to_be : int -> string` but a similar function for other verbs is missing. I don't know English well enough to know whether the general rule is easy to implement (at least for American English).

Probably the simplest to do is to work on a per message basis, with a `if String.length ...`.